### PR TITLE
refactor: change disclaimer when the nft is non-transferable

### PR DESCRIPTION
**Pull request by andreivladbrg**
Closes https://github.com/sablier-labs/v2-core/issues/818

### CODE REVIEW

**Review by smol-ninja — APPROVED** [review]
LGTM.

File: src/SablierV2NFTDescriptor.sol [hunk 1]
```diff
@@ -258,6 +260,13 @@ contract SablierV2NFTDescriptor is ISablierV2NFTDescriptor {
         pure
         returns (string memory)
     {
+        // Depending on the transferability of the NFT, declare the relevant disclaimer.
+        string memory disclaimer = isTransferable
+            ?
+            unicode"⚠️ WARNING: Transferring the NFT makes the new owner the recipient of the stream. The funds are not automatically withdrawn for the previous recipient."
+            :
+            unicode"🔒 IMPORTANT: This NFT is non-transferable. It may be listed on marketplaces but cannot be sold or transferred. Ownership is permanent.";
```
Comment (PaulRBerg) [inline src/SablierV2NFTDescriptor.sol:268]: Can we change the text to the following:

> ❕INFO: This NFT is non-transferable. It cannot be sold or transferred to another account.

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
Thanks @andreivladbrg, left some requests

File: test/unit/concrete/nft-descriptor/generateDescription.t.sol [hunk 2]
```diff
@@ -5,11 +5,13 @@ pragma solidity >=0.8.22 <0.9.0;
 import { NFTDescriptor_Unit_Concrete_Test } from "./NFTDescriptor.t.sol";
 
 contract GenerateDescription_Unit_Concrete_Test is NFTDescriptor_Unit_Concrete_Test {
-    string internal constant DISCLAIMER =
+    string internal constant INFO_TRANSFERABLE =
         unicode"⚠️ WARNING: Transferring the NFT makes the new owner the recipient of the stream. The funds are not automatically withdrawn for the previous recipient.";
+    string internal constant INFO_NON_TRANSFERABLE =
```
Comment (PaulRBerg) [inline test/unit/concrete/nft-descriptor/generateDescription.t.sol:10]: Not alphabetically ordered

File: src/SablierV2NFTDescriptor.sol [hunk 3]
```diff
@@ -248,6 +249,7 @@ contract SablierV2NFTDescriptor is ISablierV2NFTDescriptor {
 
     /// @notice Generates a string with the NFT's JSON metadata description, which provides a high-level overview.
     function generateDescription(
+        bool isTransferable,
```
Comment (PaulRBerg) [inline src/SablierV2NFTDescriptor.sol:252]: Could we please define this parameter last? After `streamId`

Comment (andreivladbrg) [reply src/SablierV2NFTDescriptor.sol:252]: after `streamId` doesn't mean last:

https://github.com/sablier-labs/v2-core/blob/e8c604733c40cb70accf0ec410cca0d785b6cbd7/src/SablierV2NFTDescriptor.sol#L250-L258

Comment (PaulRBerg) [reply src/SablierV2NFTDescriptor.sol:252]: Let's move `streamId` after `assetAddress`, and `isTransferable` after `assetAddress`

Comment (andreivladbrg) [reply src/SablierV2NFTDescriptor.sol:252]: ok 😅

**Comment by andreivladbrg** [comment]
@PaulRBerg all good now?

**Review by PaulRBerg — APPROVED** [review]
looks good now

File: src/SablierV2NFTDescriptor.sol (hunk 3 above)
Comment (smol-ninja) [reply src/SablierV2NFTDescriptor.sol]: Out of curiosity, why didn't you go with alphabetical order in this case?

Comment (PaulRBerg) [reply src/SablierV2NFTDescriptor.sol]: @smol-ninja cuz there is a natural hierarchy: contract variables first, followed by stream data.

In general: alphabetical is the default unless there's another more specific rule that makes sense in a particular case.

But this is more art than science; if you prefer, I would be open to reordering the params alphabetically.

Comment (smol-ninja) [reply src/SablierV2NFTDescriptor.sol]: Let's be artists. Gtg.

**Pull request merged**